lib/quantcast/index.js

```diff
@@ -156,38 +194,37 @@ Quantcast.prototype.completedOrder = function(track){
 };
 
 /**
- * Generate quantcast labels.
- *
- * Example:
- *
- *    options.advertise = false;
- *    labels('event', 'my event');
- *    // => "event.my event"
+ * Helper method to generate quantcast labels.
  *
- *    options.advertise = true;
- *    labels('event', 'my event');
- *    // => "_fp.event.my event"
- *
- * @param {String} type
- * @param {String} ...
+ * @param {String} prefix
+ * @param {Array} pageMeta
+ * @param {Array} ...
  * @return {String}
  * @api private
  */
 
-Quantcast.prototype._labels = function(type){
-  var args = Array.prototype.slice.call(arguments, 1);
-  var advertise = this.options.advertise;
+function createLabels(prefix, pageMeta){
+  var pageMeta = joinLabels(pageMeta, '.');
+  var labels = joinLabels([prefix, pageMeta], '.');
+  var customLabels = arguments[2];
 
-  if (advertise && 'page' == type) type = 'event';
-  if (advertise) type = '_fp.' + type;
+  if (customLabels) {
+    customLabels = joinLabels(customLabels, ',');
+    labels = joinLabels([labels, customLabels], ',');
+  }
 
-  var separator = advertise ? ' ' : '.';
-  var ret = reduce(args, function(ret, arg){
-    if (arg != null) {
-      ret.push(String(arg).replace(/, /g, ','));
-    }
-    return ret;
-  }, []).join(separator);
+  return labels.replace(/, /g, ',');
```

Comment:
This might work / read more cleanly:

```js
return join(
  compact(
    join([prefix].concat(pageMeta), '.'),
    join(customLabels, ',')
  )
, ',');
```

Test it though, I just wrote it off the top of my head lols